conjure-java-core/src/integrationInput/java/com/palantir/product/SingleUnion.java

```diff
@@ -61,6 +70,12 @@ public String toString() {
         return "SingleUnion{value: " + value + '}';
     }
 
+    private static Set<String> allKnownTypes() {
+        Set<String> types = new HashSet<>();
+        types.add("foo");
+        return types;
+    }
```

Comment:
No reason to recreate this for every invocation of `unknown`.

We could take advantage of language features above with something along these lines:

```java
public static SingleUnion unknown(String type, Object value) {
    switch (Preconditions.checkNotNull(type, "Type is required")) {
        case "foo":
            throw new SafeIllegalArgumentException(
                "Unknown type cannot be created as the provided type is known: foo");
        default:
            return new SingleUnion(new UnknownWrapper(type, Collections.singletonMap(type, value)));
    }
}
```

What do you think?